src/tools/rustbook/build.rs

```diff
@@ -0,0 +1,7 @@
+use std::env::var;
+
+fn main() {
+    if var("CI").is_err() || var("CI").is_ok() && var("CHECK_LINKS").is_ok() {
```

Comment:
Could this use parentheses to clarify the order of the operators here?

Reply:
Also I'm a bit confused as to what the purpose of this logic is and the intention of how it all plays out, can you add a comment here clarifying?

Reply:
> Could this use parentheses to clarify the order of the operators here?

`&&` are evaluated before `||` but making it more visible won't hurt.

> Also I'm a bit confused as to what the purpose of this logic is and the intention of how it all plays out, can you add a comment here clarifying?

I knew it could have been simplified but I guess my brain turned off.
It should be `var("CI").is_err() || var("CHECK_LINKS").is_ok()`
If `CI` returns error it means we are running local build and want to test links. `CHECK_LINKS` is defined in `x86_64-gnu-tools` to explicitly enable it.
I'll fix it and add comment.

Reply:
Is `CI` set on azure?  The docker builder [reexports it](https://github.com/rust-lang/rust/blame/188ab5c976a5696ac710b7ba78849ef5dcf0235a/src/ci/docker/run.sh#L170), but I don't think it is set anywhere, and AFAIK that flag won't have any effect.

Reply:
Good catch, I thought it's defined because of that reexport.